---
layout: archive
title: Zulip Chat Archive
permalink: /stream/113488-general/topic/alias.20silently.20succeeds.20when.20name.20already.20exists.html
---

<h2>Stream: <a href="https://leanprover-community.github.io/archive/stream/113488-general/index.html">general</a></h2>
<h3>Topic: <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/alias.20silently.20succeeds.20when.20name.20already.20exists.html">alias silently succeeds when name already exists</a></h3>

<hr>

<base href="https://leanprover.zulipchat.com/">

<head><link href="/style.css" rel="stylesheet"></head>

{% raw %}

<a name="247091783"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/alias%20silently%20succeeds%20when%20name%20already%20exists/near/247091783" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Floris van Doorn <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/alias.20silently.20succeeds.20when.20name.20already.20exists.html#247091783">(Jul 24 2021 at 20:31)</a>:</h4>
<p><span class="user-mention" data-user-id="110049">@Mario Carneiro</span> is this intended? If you use <code>alias</code> and the declarations already exist, then alias does not raise an error.</p>
<div class="codehilite" data-code-language="Lean"><pre><span></span><code><span class="kn">import</span> <span class="n">tactic.alias</span>

<span class="n">alias</span> <span class="n">iff.refl</span> <span class="bp">â†”</span> <span class="n">rfl</span> <span class="n">eq.refl</span> <span class="c1">-- no error, but does nothing</span>
</code></pre></div>



<a name="247091835"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/alias%20silently%20succeeds%20when%20name%20already%20exists/near/247091835" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Floris van Doorn <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/alias.20silently.20succeeds.20when.20name.20already.20exists.html#247091835">(Jul 24 2021 at 20:32)</a>:</h4>
<p>This came up in a PR where I added a new lemma, which broke a later alias command.</p>



<a name="247111453"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/alias%20silently%20succeeds%20when%20name%20already%20exists/near/247111453" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Mario Carneiro <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/alias.20silently.20succeeds.20when.20name.20already.20exists.html#247111453">(Jul 25 2021 at 06:20)</a>:</h4>
<p><span class="user-mention" data-user-id="111080">@Floris van Doorn</span> I don't really remember, and both possibilities seem acceptable to me. I can imagine that not erroring is sometimes helpful for autogenerated lemmas like in <code>to_additive</code>, but for explicit use erroring might help avoid typos. If you want to change it to give an error and fix any existing uses, be my guest</p>



{% endraw %}

<hr><p>Last updated: Jan 25 2023 at 00:06 UTC</p>